Realm/RLMRealm.mm

```diff
@@ -944,16 +945,24 @@ - (RLMObject *)createObject:(NSString *)className withValue:(id)value {
 }
 
 - (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(NSData *)key error:(NSError **)error {
+    return [self writeCopyToURL:fileURL encryptionKey:key enableHistory:NO error:error];
+}
+
+- (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(NSData *)key enableHistory:(BOOL)shouldSync error:(NSError **)error {
     key = RLMRealmValidatedEncryptionKey(key);
     NSString *path = fileURL.path;
 
     try {
-        _realm->verify_thread();
-        try {
-            _realm->read_group().write(path.UTF8String, static_cast<const char *>(key.bytes));
-        }
-        catch (...) {
-            _impl::translate_file_exception(path.UTF8String);
+        if (shouldSync) {
```

Comment:
It does not make sense to wait for uploads because the Realm is always currently up to date (w.r.t data it is going to upload). The only argument is that we should wait for uploads to make sure that the sync history is full up to date before copy, but this would then make `writeCopy` a blocking call and would require network access.